vector-stores/spring-ai-gemfire/src/main/java/org/springframework/ai/vectorstore/GemFireVectorStore.java

```diff
@@ -161,23 +176,55 @@ public GemFireVectorStore(GemFireVectorStoreConfig config, EmbeddingClient embed
 
 	private static final class CreateRequest {
 
-		@JsonProperty("embedding-type")
-		private final String embeddingType = "float";
+		private String name;
+
+		@JsonProperty("beam-width")
+		private int beamWidth = 100;
+
+		@JsonProperty("max-connections")
+		private int maxConnections = 16;
+
+		@JsonProperty("vector-similarity-function")
+		private String vectorSimilarityFunction = "COSINE";
+
+		private final String[] fields = new String[] { "vector" };
+
+		private final int buckets = 0;
```

Comment:
should this be configurable?

Reply:
I have temporarily removed these fields. If need be, we can explore configuring them within the scope of this PR.

Reply:
Shouldn't we expose all the fields the endpoint exposes?  So buckets and fields should be exposed.  Unless there is a spring AI reason not to